pkg/processor/cascadingfilterprocessor/sampling/drop_trace_factory.go

```diff
@@ -104,6 +112,15 @@ func (dte *dropTraceEvaluator) ShouldDrop(_ pcommon.TraceID, trace *TraceData) b
 						matchingNumericAttrFound = checkIfNumericAttrFound(span.Attributes(), dte.numericAttr)
 					}
 
+					isRootSpan := span.ParentSpanID().IsEmpty()
+
+					if dte.statusCode != nil && !matchingStatusCodeFound && isRootSpan {
```

Comment:
```suggestion
					if !matchingStatusCodeFound && dte.statusCode != nil && span.ParentSpanID().IsEmpty() {
```

There is no need to run `span.ParentSpanID().IsEmpty()` for every span, so I would move it after checking for status code

Reply:
Also I moved `!matchingStatusCodeFound` to the beginning as this is more logical order

Reply:
yea that makes sense